shared/profile/index.js

```diff
@@ -27,67 +35,92 @@ class Profile extends Component<void, Props, void> {
+
+export default connector.connect((state, dispatch, ownProps) => {
+  const stateProps = {
     myUsername: state.config.username,
     trackers: state.tracker.trackers,
-  }),
-  dispatch => ({
-    refresh: username => {
-      dispatch(getProfile(username))
-      dispatch(updateTrackers(username))
-    },
+  }
+
+  const username = ownProps.username || stateProps.myUsername || ''
+
+  const {getProfile, updateTrackers, onFollow, onUnfollow} = trackerActions
+
+  const refresh = () => {
+    dispatch(getProfile(username))
+    dispatch(updateTrackers(username))
+  }
+
+  const dispatchProps = {
     onUserClick: username => { dispatch(routeAppend({path: 'profile', username})) },
-    onBack: () => dispatch(navigateUp()),
-    onFolderClick: folder => dispatch(openInKBFS(folder.path)),
-    onEditProfile: () => dispatch(routeAppend({path: 'editprofile'})),
-  }),
-  (stateProps, dispatchProps, ownProps) => {
-    const username = ownProps.username || stateProps.myUsername
-    const isYou = username === stateProps.myUsername
-    const onEditProfile = () => dispatchProps.onEditProfile()
-    const bioEditFns = isYou && {
-      onBioEdit: onEditProfile,
-      onEditAvatarClick: onEditProfile,
-      onEditProfile: onEditProfile,
-      onLocationEdit: onEditProfile,
-      onNameEdit: onEditProfile,
-      onMissingProofClick: () => console.log('TODO onMissingProofClick'),
-    }
+    onBack: () => { dispatch(navigateUp()) },
+    onFolderClick: folder => { dispatch(openInKBFS(folder.path)) },
+    onEditProfile: () => { dispatch(routeAppend({path: 'editprofile'})) },
+    onFollow: () => { dispatch(onFollow(username, false)) },
+    onUnfollow: () => { dispatch(onUnfollow(username)) },
+    onAcceptProofs: () => { dispatch(onFollow(username, false)) },
+  }
 
-    return {
-      ...ownProps,
-      ...stateProps.trackers[username],
-      ...dispatchProps,
-      isYou,
-      bioEditFns,
-      username,
-      refresh: username => dispatchProps.refresh(username),
-      loading: isLoading(stateProps.trackers[username]),
-    }
+  const onMissingProofClick = () => { console.log('TODO onMissingProofClick') }
+
+  const isYou = username === stateProps.myUsername
+  const onEditProfile = () => { dispatchProps.onEditProfile() }
+  const bioEditFns = isYou ? {
+    onBioEdit: onEditProfile,
+    onEditAvatarClick: onEditProfile,
+    onEditProfile: onEditProfile,
+    onLocationEdit: onEditProfile,
+    onNameEdit: onEditProfile,
+    onMissingProofClick,
+  } : null
+
+  const trackerState = stateProps.trackers[username]
+
+  if (trackerState.type !== 'tracker') {
+    console.warn('Expected a tracker type, trying to show profile for non user')
+    return null
+  }
+
+  return {
+    ...ownProps,
+    ...trackerState,
+    ...dispatchProps,
+    isYou,
+    bioEditFns,
+    username,
+    refresh,
+    followers: trackerState.trackers || [],
```

Comment:
Still necessary to `|| []` now that you moved default values into the reducer initial state?